### ISSUE TITLE
Quiz: Ranking component

### ISSUE THREAD

**Issue by sheepyhollow** [zentrumnawi/solid-frontend#744]
The component described in #708 should be enhanced to

a) work properly ;-)
b) visualize a "semi-correct" answer if some items are in correct order realtive to one another but not in their correct slot.

The behaviour should be optional - there will be a flag/qualifier in the endpoint if the input is to be evaluated in this way.

[ ] develop an algorithm to check for correct sequences
[ ] find a way to visualize "subsequences"

**Comment by sheepyhollow** [comment]
I will close the original issue #708

**Comment by sheepyhollow** [comment]
The data model will yet have to be written but it should look like this:

- question_text(String/Markdown)
- Items (array)
  - Item Text (string/markdown)
  - correct position (number)
- feedback correct (string/markdown) 
- feedback wrong (string/markdown)
- evaluate subsequences (boolean)
- feedback subsequences (string/markdown)
- tag
- difficulty

**Comment by sheepyhollow** [comment]
ah - and please rename the component in your quiz/abstraction branch to `ranking-question`

**Comment by Haseeb715** [comment]
Yeah sure. I will rename and organize the code as well.

**Comment by sheepyhollow** [comment]
Erklärung zum Answer Model:

_Jede_ Antwort enthält
* text
* position
* Feedback correct
* Feedback wrong
* Subsequences (Boolean)
* Feedback Subsequences

Wir brauchen die letzten 4 Felder aber nur _einmal_ pro Frage.

Vorschlag: Es wird immer die Information aus der Antwort mit position = 1 genommen.
Das muss auch das Content-Team wissen.

**Comment by lthuong** [comment]
Sollen wir die Anzahl der validen Subsequences auf 1 setzen, also wir nehmen nur die längste ?

**Comment by sheepyhollow** [comment]
Ja, das würde ich vorschlagen. Wir haben ja auch noch nicht genau festgelegt, wie das visuelle Feedback aussieht, oder?
Da oben sind ja nur Vorschläge..

Hast Du schon einen Algorithmus für die Subsequences? Ich habe mir noch gar keine Gedanken dazu gemacht - ich dachte mir immer, dass das eine tolle Aufgabe für Informatiker ist 😆 (no pressure, though)

**Comment by sheepyhollow** [comment]
Ich würde übrigens noch kleine Animations für den Drag/Drop-Effekt schön finden - so wie im [Beispiel von material.angular.io](https://material.angular.io/cdk/drag-drop/overview#reordering-lists). Das steht dort im CSS, glaube ich.

**Comment by lthuong** [comment]
Also ich finde, dass Farbe und Symbole einen guten Vorschlag ist. Ich probiere erstmal aus, damit wir zusammen sehen können wie es aussieht. 

Yup ich habe schon Pseudecode geschrieben und muss den nur noch implementieren. Ich finde die ist doch eine interessante Aufgabe 👍
